topology.go

```diff
@@ -11,11 +11,13 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// Topology defines the baker topology, that is how to retrieve records (input),
+// how to process them (filter), and where to output the results (output+upload)
 type Topology struct {
-	Input   Input
-	Filters []Filter
-	Output  []Output
-	Upload  Upload
+	Input   Input    // The input component
+	Filters []Filter // The list of filters
+	Output  []Output // The output component (it's a slice because can be sharded)
+	Upload  Upload   // The optional upload component
```

Comment:
This is a case where the convention would be to write something like:

Input is the input component
Filters are the filters components
etc.

In this case it's admitted to not document the fields and that's what I think we could do since the comments wouldn't add any information.

however the Output comment is not correct. It's not a slice because it can be sharded, it's a slice because you can set procs>1 so I think we could ommit the comment also here, as you prefer